ui/js/controllers/tcjobactions.js

```diff
@@ -0,0 +1,95 @@
+"use strict";
+
+treeherder.controller('TCJobActionsCtrl', [
+    '$scope', '$http', '$uibModalInstance', 'ThResultSetStore',
+    'ThJobDetailModel', 'thTaskcluster', 'ThTaskclusterErrors',
+    'thNotify', 'job', 'repoName', 'resultsetId',
+    function($scope, $http, $uibModalInstance, ThResultSetStore,
+             ThJobDetailModel, thTaskcluster, ThTaskclusterErrors, thNotify,
+             job, repoName, resultsetId) {
+        let originalTask;
+        $scope.input = {};
+
+        $scope.cancel = function() {
+            $uibModalInstance.dismiss('cancel');
+        };
+
+        $scope.updateSelectedAction = function() {
+            $scope.input.jsonPayload = JSON.stringify(jsonSchemaDefaults($scope.input.selectedAction.schema), null, 4);
+        };
+
+        $scope.triggerAction = function() {
+            $scope.triggering = true;
+
+            let task = _.clone($scope.input.selectedAction.task);
+            // update the various date properties with evaluations
+            task = thTaskcluster.refreshTimestamps(task);
+            // fixme: it seems like action tasks define these properties as
```

Comment:
I think the confusion here is that `refreshTimestamps` was created for `action.yml` rather than `action.json`. I expect we'll want to do something different with the json version since it has the `$fromNow` stuff in it.